src/main/java/org/prebid/server/auction/PrivacyEnforcementService.java

```diff
@@ -0,0 +1,272 @@
+package org.prebid.server.auction;
+
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Geo;
+import com.iab.openrtb.request.Regs;
+import com.iab.openrtb.request.User;
+import io.vertx.core.Future;
+import io.vertx.core.json.Json;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.auction.model.PrivacyEnforcementResult;
+import org.prebid.server.bidder.BidderCatalog;
+import org.prebid.server.execution.Timeout;
+import org.prebid.server.gdpr.GdprService;
+import org.prebid.server.gdpr.model.GdprResponse;
+import org.prebid.server.metric.Metrics;
+import org.prebid.server.proto.openrtb.ext.request.ExtRegs;
+import org.prebid.server.proto.openrtb.ext.request.ExtUser;
+import org.prebid.server.proto.response.BidderInfo;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Service provides masking for User, Device, Regs.
+ */
+public class PrivacyEnforcementService {
+
+    private static final DecimalFormat ROUND_TWO_DECIMALS =
+            new DecimalFormat("###.##", DecimalFormatSymbols.getInstance(Locale.US));
+
+    private final GdprService gdprService;
+    private final BidderCatalog bidderCatalog;
+    private final Metrics metrics;
+    private final boolean useGeoLocation;
+
+    public PrivacyEnforcementService(GdprService gdprService, BidderCatalog bidderCatalog, Metrics metrics,
+                                     boolean useGeoLocation) {
+        this.gdprService = gdprService;
+        this.bidderCatalog = bidderCatalog;
+        this.metrics = metrics;
+        this.useGeoLocation = useGeoLocation;
+    }
+
+    /**
+     * Returns {@link Future &lt;{@link Map}&lt;{@link String}, {@link PrivacyEnforcementResult }&gt;&gt;}, where
+     * bidders name mapped to masked {@link PrivacyEnforcementResult}
+     */
+    Future<Map<String, PrivacyEnforcementResult>> mask(Map<String, User> bidderToUser, ExtUser extUser, ExtRegs extRegs,
+                                                       List<String> bidders, Map<String, String> aliases,
+                                                       BidRequest bidRequest, String publisherId, Timeout timeout) {
+        final Regs regs = bidRequest.getRegs();
+        final boolean coppaMasking = isCoppaMaskingRequired(regs);
```

Comment:
Pls consider to move coppa masking requirement to `getBidderToPrivacyEnforcementResult(...)` - near to using of it. Since we don't decrease performance in anycase.